shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-distsql/shardingsphere-sharding-distsql-handler/src/main/java/org/apache/shardingsphere/sharding/distsql/handler/update/DropShardingAlgorithmStatementUpdater.java

```diff
@@ -0,0 +1,107 @@
+        checkShardingAlgorithmsExisted(schemaName, sqlStatement, currentRuleConfig);
+        checkShardingAlgorithmsInUsed(schemaName, sqlStatement, currentRuleConfig);
+    }
+    
+    private void checkCurrentRuleConfiguration(final String schemaName, final ShardingRuleConfiguration currentRuleConfig) throws RequiredAlgorithmMissedException {
+        if (null == currentRuleConfig) {
+            throw new RequiredAlgorithmMissedException(schemaName);
+        }
+    }
+    
+    private void checkShardingAlgorithmsExisted(final String schemaName, final DropShardingAlgorithmStatement sqlStatement,
+                                                final ShardingRuleConfiguration currentRuleConfig) throws RequiredAlgorithmMissedException {
+        Collection<String> currentShardingAlgorithms = getCurrentShardingAlgorithms(currentRuleConfig);
+        Collection<String> notExistedAlgorithms = sqlStatement.getAlgorithmNames().stream().filter(each -> !currentShardingAlgorithms.contains(each)).collect(Collectors.toList());
+        if (!notExistedAlgorithms.isEmpty()) {
+            throw new RequiredAlgorithmMissedException(schemaName, notExistedAlgorithms);
+        }
+    }
+    
+    private void checkShardingAlgorithmsInUsed(final String schemaName, final DropShardingAlgorithmStatement sqlStatement, final ShardingRuleConfiguration currentRuleConfig) 
+            throws AlgorithmInUsedException {
+        Collection<String> allInUsed = getAllOfAlgorithmsInUsed(currentRuleConfig);
+        Collection<String> usedAlgorithms = sqlStatement.getAlgorithmNames().stream().filter(allInUsed::contains).collect(Collectors.toList());
+        if (!usedAlgorithms.isEmpty()) {
+            throw new AlgorithmInUsedException(schemaName, usedAlgorithms);
+        }
+    }
+    
+    private Collection<String> getAllOfAlgorithmsInUsed(final ShardingRuleConfiguration shardingRuleConfig) {
+        Collection<String> result = new LinkedHashSet<>();
+        shardingRuleConfig.getTables().stream().forEach(each -> {
+            if (null != each.getDatabaseShardingStrategy()) {
+                result.add(each.getDatabaseShardingStrategy().getShardingAlgorithmName());
+            }
+            if (null != each.getTableShardingStrategy()) {
+                result.add(each.getTableShardingStrategy().getShardingAlgorithmName());
+            }
+        });
+        return result;
+    }
+    
+    private Collection<String> getCurrentShardingAlgorithms(final ShardingRuleConfiguration shardingRuleConfig) {
+        return shardingRuleConfig.getShardingAlgorithms().keySet();
+    }
+    
+    @Override
+    public boolean updateCurrentRuleConfiguration(final DropShardingAlgorithmStatement sqlStatement, final ShardingRuleConfiguration currentRuleConfig) {
+        for (String each : sqlStatement.getAlgorithmNames()) {
+            dropShardingAlgorithm(currentRuleConfig, each);
+        }
+        return false;
```

Comment:
should real result be returned here?

Reply:
refer to `DropShardingTableRuleStatementUpdater` and `DropShardingBindingTableRuleStatementUpdater`